src/sentry/static/sentry/app/views/alerts/rules/filter.tsx

```diff
@@ -0,0 +1,185 @@
+
+type DropdownButtonProps = React.ComponentProps<typeof DropdownButton>;
+
+export type RenderProps = {
+  toggleFilter: (filter: string) => void;
+};
+
+type RenderFunc = (props: RenderProps) => React.ReactElement;
+
+type Props = {
+  header: string;
+  onFilterChange: (filterSelection: Set<string>) => void;
+  filterList: string[];
+  children: RenderFunc;
+  selection: Set<string>;
+};
+
+class Filter extends React.Component<Props> {
+  toggleFilter = (filter: string) => {
+    const {onFilterChange, selection} = this.props;
+    const newSelection = new Set(selection);
+    if (newSelection.has(filter)) {
+      newSelection.delete(filter);
+    } else {
+      newSelection.add(filter);
+    }
+    onFilterChange(newSelection);
+  };
+
+  toggleAllFilters = () => {
+    const {filterList, onFilterChange, selection} = this.props;
+    const newSelection =
+      selection.size === filterList.length
+        ? (new Set() as Set<string>)
+        : new Set(filterList);
+
+    onFilterChange(newSelection);
+  };
+
+  getNumberOfActiveFilters = (): number => {
+    const {selection} = this.props;
+    return selection.size;
+  };
+
+  render() {
+    const {children, header, filterList} = this.props;
+    const checkedQuantity = this.getNumberOfActiveFilters();
+
+    const dropDownButtonProps: Pick<DropdownButtonProps, 'children' | 'priority'> & {
+      hasDarkBorderBottomColor: boolean;
+    } = {
+      children: (
+        <React.Fragment>
+          <IconFilter size="xs" />
+          <FilterLabel>{t('Filter')}</FilterLabel>
+        </React.Fragment>
+      ),
+      priority: 'default',
+      hasDarkBorderBottomColor: false,
+    };
+
+    if (checkedQuantity > 0) {
+      dropDownButtonProps.children = (
+        <span>{tn('%s Active Filter', '%s Active Filters', checkedQuantity)}</span>
+      );
+      dropDownButtonProps.priority = 'primary';
+      dropDownButtonProps.hasDarkBorderBottomColor = true;
+    }
+    return (
+      <DropdownControl
+        menuWidth="240px"
+        blendWithActor
+        button={({isOpen, getActorProps}) => (
+          <StyledDropdownButton
+            {...getActorProps()}
+            showChevron={false}
+            isOpen={isOpen}
+            hasDarkBorderBottomColor={dropDownButtonProps.hasDarkBorderBottomColor}
+            priority={dropDownButtonProps.priority as DropdownButtonProps['priority']}
+            data-test-id="filter-button"
+          >
+            {dropDownButtonProps.children}
+          </StyledDropdownButton>
+        )}
+      >
+        <Header>
+          <span>{header}</span>
+          <CheckboxFancy
+            isChecked={checkedQuantity > 0}
+            isIndeterminate={checkedQuantity > 0 && checkedQuantity !== filterList.length}
+            onClick={event => {
+              event.stopPropagation();
+              this.toggleAllFilters();
+            }}
+          />
+        </Header>
+        {children({toggleFilter: this.toggleFilter})}
```

Comment:
This can be replaced with the map render of `ListItem`s  in `rules/index.tsx`, ref the comment below

Reply:
mmm yea I recognize that the filter component doesn't have to be a renderprops, but I was thinking that in the future we'd want to abstract the filter component entirely. Which would mean a parent component would be in charge of determining the children that live inside the filter menu. If you think it's too early to do that I can move the `ListItem`s in for now and then when I end up abstracting the filter component I can worry about the renderprops